gems/smithy/lib/smithy/anvil/client/templates/shapes.erb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module <%= namespace %>
+  module Shapes
+    include Smithy::Client::Shapes
+
+<%# Shapes definition -%>
+  <%- @shapes.each do |shape| -%>
+    <%- params = ["id: \"#{shape.id}\""] -%>
+    <%- params << "type: Types::#{shape.name}" if shape.typed -%>
+    <%- params << "traits: #{shape.traits}" unless shape.traits.empty? -%>
+    <%= "#{shape.name} = #{shape.shape_type}.new(" %><%= params.join(', ') %>)
+  <%- end -%>
+<%# Shape members definition %>
+  <%- shapes_with_members.each do |shape| -%>
+    <%- shape.members.each do |member| -%>
+    <%- params = [] -%>
+    <%- params << ["\"#{member.name}\""] unless %w[ListShape MapShape].include? shape.shape_type -%>
+    <%- params << [member.shape] -%>
+    <%- params << "traits: #{member.traits}" unless member.traits.empty? -%>
+    <%- case shape.shape_type -%>
+    <%- when 'ListShape' -%>
+    <%= "#{shape.name}.set_member(" %><%= params.join(', ') %>)
+    <%- when 'MapShape' -%>
+    <%= "#{shape.name}.set_member_#{member.name}(" %><%= params.join(', ') %>)
+    <%- else -%>
+    <%= "#{shape.name}.add_member(" %><%= params.join(', ') %>)
+    <%- end -%>
+    <%- end -%>
+  <%- end -%>
+<%# Schema definition %>
+    SCHEMA = Smithy::Client::Schema.new do |schema|
+      schema.service = ServiceShape.new(
+        id: "<%= @service_shape.id %>",
+        version: "<%= @service_shape.version %>",
+        traits: <%= @service_shape.traits %>
+      )
+    <%- @operation_shapes.each do |shape| %>
+      schema.add_operation(:<%= shape.name %>, OperationShape.new do |operation|
+        operation.id = "<%= shape.id %>"
+        operation.input = <%= shape.input %>
+        operation.output = <%= shape.output %>
+       <%- unless shape.traits.empty? -%>
```

Comment:
I don't think we care to check this. Have it set a hash or nil to this method.

Reply:
Yeah I was torn whether we want to render empty hashes or not.

Reply:
I don't understand. Either an empty hash or nil will be assigned here. We wouldn't want an empty hash.

Reply:
My intention was that if `OperationShape` is created without a traits param, it will default to an empty hash. Therefore, here I want to check to see if the traits exists to worth rendering

Reply:
I think you can simply just assign shape.traits, and shape.traits will either be a hash with stuff or nil.